aspnetcore/blazor/hybrid/tutorials/maui-blazor-web-app.md

```diff
@@ -0,0 +1,282 @@
+
+```razor
+@using static InteractiveRenderSettings
+```
+
+For the Blazor Web App on the web client, the property values are assigned from <xref:Microsoft.AspNetCore.Components.Web.RenderMode>. When the components are loaded into a <xref:Microsoft.AspNetCore.Components.WebView.Maui.BlazorWebView> for the MAUI project's native client, the render mode is unassigned (`null`) because the MAUI project explicitly sets the render mode properties to `null` when `ConfigureBlazorHybridRenderModes` is called.
+
+At this point, you can adopt either:
+
+* [Global interactive rendering](#global-interactive-rendering)
+* [Per-page/component rendering](#per-pagecomponent-rendering)
+
+### Global interactive rendering
+
+The MAUI project is interactive by default, so no action is taken at the project level in the MAUI project other than calling `InteractiveRenderSettings.ConfigureBlazorHybridRenderModes`.
+
+The Blazor Web App sets a global interactive render mode normally in the `App` component (`Components/App.razor`). In the following example, the `HeadOutlet` and `Routes` components assign `InteractiveServer` from <xref:Microsoft.AspNetCore.Components.Web.RenderMode> via the Blazor Web App's `@using static Microsoft.AspNetCore.Components.Web.RenderMode` directive in the project's `_Imports` file:
+
+```razor
+<HeadOutlet @rendermode="InteractiveServer" />
+
+...
+
+<Routes @rendermode="InteractiveServer" />
+```
+
+At this point, no component in the RCL sets a render mode. A component's render mode is inherited from the global setting.
+
+To recap:
+
+* For components rendered by MAUI for the native client, RCL component render modes are `null`. The components are interactive by default.
+* For components rendered by the Blazor Web App for the web client, RCL component render modes are inherited from the global setting in the `App` component.
+
+### Per-page/component rendering
+
+The MAUI project is interactive by default, so no action is taken at the project level in the MAUI project other than calling `InteractiveRenderSettings.ConfigureBlazorHybridRenderModes`.
+
+Don't set an `@rendermode` directive attribute on the `HeadOutlet` and `Routes` components of the Blazor Web App's `App` component (`Components/App.razor`):
+
+```razor
+<HeadOutlet />
+
+...
+
+<Routes />
+```
+
+Apply render modes to individual components in the RCL.
+
+The Razor class library (RCL) contains a global static `@using` directive (`@using static InteractiveRenderSettings`) to reference `InteractiveRenderSettings` class properties in the RCL's components. A component adopts a given render mode by assigning the render mode from `InteractiveRenderSettings` at the top of its component definition file (`.razor`) under its `@page` directive:
+
+Interactive Server:
+
+```razor
+@rendermode InteractiveServer
+```
+
+Interactive Auto:
+
+```razor
+@rendermode InteractiveAuto
+```
+
+Interactive WebAssembly:
+
+```razor
+@rendermode InteractiveWebAssembly
```

Comment:
I think it's important to call out that if you want to share your UI with MAUI but also specify a per page InteractiveWebAssembly setting you'll want two RCLs, one with the wasm-only components and one with the rest. I show how to set that up in this branch: https://github.com/BethMassi/HybridSharedUI/tree/WebAsm-PPC

Also, anytime one selects the interactivity as Auto or WebAssembly then you also get a Web.Client project in your solution no matter the location.

Reply:
Cool ... it looks like after an edit pass that these are the only items that I'll need to address.

Reply:
It's a bit of a *Wowzers!* complicated setup.

It's going to take a bit more time to roll that into the mix. I now need to see if I need an entirely new section or if I can roll those aspects (WRT the 2nd RCL) into the existing *Per-component* section.

I still hope to have it drafted up today by this afternoon.

Reply:
I'm trying to do it in the *Per-component* subsection now.

I just pushed a commit. I need to step out OOF for a couple of hours, but I'll be back soon to work on it a bit more. You can look at the latest commit (or the whole DIFF) to see the latest updates that include where/how I addressed the Auto/WASM components and 2nd RCL setup.